### ISSUE TITLE
WCAG 2.4.5 ✅ (Partially Supports): Multiple Ways

### ISSUE THREAD

**Issue by StephDriver** [openlibhums/janeway#5503]
**Codebase commit:** 1.9.x 91c7eb9e7
**Result:** *supports*

### Resources
[WCAG 2.2 Success Criterion](https://www.w3.org/WAI/WCAG22/Understanding/multiple-ways.html)

>More than one way is available to locate a [web page](https://www.w3.org/WAI/WCAG22/Understanding/multiple-ways.html#dfn-web-page) within a [set of web pages](https://www.w3.org/WAI/WCAG22/Understanding/multiple-ways.html#dfn-set-of-web-pages) except where the web page is the result of, or a step in, a [process](https://www.w3.org/WAI/WCAG22/Understanding/multiple-ways.html#dfn-process).

### Audit Checklist
- [x] Review previous Audit Result
- [x] Review work identified in previous audit and list open issues.
- [x] Check for any other relevant open issues.
- [x] Retest open issues, are these issues still present?
- [x] Test for any new issues that may have been introduced.
- [x] Determine compliance status, add this to issue name.
- [x] Add any new issues to the Accessibility project for triage.

### Previous audit
#5150

> 🟠 Partially Supports

> ✅ Articles are supported. They can be reached via the articles page (which includes the search) and via the issue in which they appeared. If keyword linking is turned on for that journal, then the keywords for each article link to a list of other articles with the same keyword. Articles that are part of special collections can also be reached via their collection.

> 🟠 News items are partially supported. All news is reachable via the news page. News items have optional tags, and these are links to other items with the same tag, however news items could have no tags at all, or have a tag that is unique.

> ❌ Static pages are not supported. They are reachable via the navigation and some may be linked from the contents of the homepage, but that is configurable by journal. Some could cross-reference each other via links, but that would be in the user-content.

> The sitemaps should have provided the second way for all missing support above, but are themselves not working as expected (see tests below) and so cannot be relied upon.

#### Work identified in last audit

- https://github.com/openlibhums/janeway/issues/5170

 ### Audit Notes
There are four categories of pages within Janeway.
1. publications (e.g. articles, issues, preprints, collections)
2. news items
3. static built-in (e.g. search pages, contact us, accessibility, homepage, sitemaps)
4. static user-managed (the CMS pages which users create)

Taking these one by one.
1. ✅ publications - once published these are reachable via the navigation in multiple ways, as previously described in the last audit. These pages were already compliant.
2. ✅ news - similar to publications, these were previously all reachable via the navigation, however whether they were reachable in a second way depended on whether they were tagged. This was user content, and not within our control.  Adding news items to the sitemaps provides a second way for each to be reached, and so news pages are now compliant.
3. ✅ static built-in pages - These are navigable from the navigation or the footer. Previously these were not reliably included in the sitemaps. The updated sitemap generation includes them.
4. ✅ static CMS pages - These were not included in sitemaps previously and so were only reachable via user-entered in-page links or the user adding them to the navigation.   It would have been possible for a user to have ensured all had more than one way to reach them, however there was nothing in Janeway to facilitate this.  CMS pages may now be draft or published.  Investigation found that users had been creating drafts by publishing pages and not linking to them, which on installs with a large number of CMS pages made it difficult to keep track of what was meant to be reached by users and what should not. Now it is clear which are draft and which are published so the users should be clear about which pages to include in navigation. All published CMS pages are added to the sitemaps automatically.

All that remains is to check the sitemaps generator is doing what it claims.
Results below - TLDR, there is one type of page that is not compliant.

#### Conclusion 

✅ Supports.

Support for multiple ways is theme independent. The same sitemaps are shown regardless of theme.

 ### Work arising
 - #5529

**Comment by StephDriver** [comment]
## Testing

I have created two automated tests:
1. a management command that grabs all the URLs from the generated sitemaps that aren't links to other sitemaps. 
2. a playwright crawler that crawls for any links in the non-login pages (i.e. front of house) and matches them against the url list above.

I have the crawler excluding the following:
- download links, theses are not "pages".
- likewise 'print' links. 
- paginations (e.g. for search results, page 1, page 2, page 3 etc are user-configurable ways of displaying results. The main page should be on the sitemap.)
- news tags. These are filters for news items, again not pages for this purpose.
- article keywords - likewise to news tags.

The sitemaps include some login-gated pages, as they are likely to be ones users want to find, so these are exceptions in the other direction where I would expect the sitemap to have entries but would not show up on the crawler.
- The Reviewer landing page: http://<domain>/reviewer/
- The Submissions landing page: http://<domain>/submit/start/

**Comment by StephDriver** [comment]
Further testing - crawler found only one link to `accepted, not yet published` article pages. Sitemaps don't cover them. So multiple ways not achieved.

These articles are excluded from the articles page (i.e the search) as that's just published articles.

By default the aren't shown anywhere, however they can be included in featured articles on the homepage, or linked manually by journal staff. So it's quite possible most installs will be compliant because they aren't linked anywhere.

**Comment by StephDriver** [comment]
@ajrbyers Pausing on closing this audit, because the work is being considered for rapid development.  Will either close it and mark the result or update once that is known.

**Comment by StephDriver** [comment]
Following discussion within the team:

1. unpublished article are deliberately excluded from other front of house links (e.g. search results, issue lists). So they are not expected to be linked on the front of house, not expected to be reachable at all.
2. The test data I was using had an unpublished article linked to the featured articles on the homepage. This seems an unlikely real-world situation, but more a 'generated test data' scenario.

At this point, while not mathematically proven, it is reasonable to consider janeway compliant based on all the other pages having 'multiple ways' in at least one from front-of-house and one from sitemaps. 

Updating to call this compliant.
